mock/mock.go

```diff
@@ -1006,6 +1007,13 @@ func (args Arguments) Diff(objects []interface{}) (string, int) {
 			switch expected := expected.(type) {
 			case anythingOfTypeArgument:
 				// type checking
+				if actual == nil {
+					if expected != "<nil>" {
+						differences++
+						output = fmt.Sprintf("%s\t%d: FAIL:  type %s != type <nil> - %s\n", output, i, expected, actualFmt)
```

Comment:
<nil> is not a type itself


```suggestion
						output = fmt.Sprintf("%s\t%d: FAIL:  type %s != <nil> - %s\n", output, i, expected, actualFmt)
```

Reply:
This was to match the format of the other failure case, it should be kept to disambiguate type <nil> from value <nil>. I have now deleted this line, though.